common/core/gpudevice.cpp

```diff
@@ -61,9 +63,10 @@ bool GpuDevice::Initialize() {
   initialization_state_ &= ~kHWCSettingsDone;
   thread_sync_lock_.unlock();
 
-  if (use_thread && lock_fd_ == -1) {
+  if (-1 == lock_fd_) {
```

Comment:
I see use_thread variable has chance to be set to false from code logic. Remove this is to ensure HWCThread::Exit is called if the /vendor/hwc.lock open returns -1. After double reviewed code logic, I believe there is no risk if revert this little change since the thread will be fall into waiting status in HandleWait if lock_fd_ equals -1. I will revert this piece of change and submit a new patch.